src/sentry/preprod/api/endpoints/project_preprod_artifact_image.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import annotations
+
+import logging
+
+from django.http import HttpResponse
+from rest_framework.request import Request
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.models.project import Project
+from sentry.objectstore import get_preprod_session
+
+logger = logging.getLogger(__name__)
+
+
+@region_silo_endpoint
+class ProjectPreprodArtifactImageEndpoint(ProjectEndpoint):
+    owner = ApiOwner.EMERGE_TOOLS
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+    }
+
+    def get(
+        self,
+        _: Request,
+        project: Project,
+        image_id: str,
+    ) -> HttpResponse:
+
+        organization_id = project.organization_id
+        project_id = project.id
+
+        object_key = f"{organization_id}/{project_id}/{image_id}"
+        session = get_preprod_session(organization_id, project_id)
+
+        try:
+            result = session.get(object_key)
+            # Read the entire stream at once (necessary for content_type)
+            image_data = result.payload.read()
```

Comment:
We have a generic endpoint that proxies through to objectstore now (`OrganizationObjectstoreEndpoint`). What if you issue a temporary redirect (307) instead and let that endpoint do the proxying? This will also include the content-type header.

@lcian we should add a `get_proxy_url` to `sentry.objectstore` similar to `get_symbolicator_url` that outputs the route. That allows us to add the signature once we ship it.

Reply:
Also we can just leverage the generic endpoint from our frontend use-case, but also will suffer from the same limitations of the feature flag here, so once that's enabled for a wider audience I can make that change and just likely remove this endpoint entirely.